slot.go

```diff
@@ -144,17 +144,23 @@ func (y Yubikey) KeyManagement() (*Slot, error) {
 // Get a Slot off of the Yubikey by the SlotId.
 //
 // This will trigger an attempt to get (and parse) the x509 Certificate
-// for this slot. Only slots with an x509 Certificate can be used.
+// for this slot. Only slots with an x509 Certificate can be used on
+// firmware < 4.3. On firmware >= 4.3 any slots with a private key generated
+// on the Yuibkey can be used if attestation key is present.
 func (y Yubikey) Slot(id SlotId) (*Slot, error) {
 	/* Right, let's see what we can do here */
 	slot := Slot{yubikey: y, Id: id}
 
 	certificate, err := slot.GetCertificate()
 	if err != nil {
-		return nil, err
+		var errAttest error
+		certificate, errAttest = y.Attest(id)
+		if errAttest != nil {
+			return nil, err
```

Comment:
since this doesn't break old behavior or older tokens I think I'm tempted to take it.

Reply:
This won't work for imported keys, but it won't work for imported keys today, either so this only helps I think